### PR TITLE
Translations update from Weblate

### PR DESCRIPTION
**Pull request by weblate-fpf**
Translations update from [Weblate](https://weblate.securedrop.org) for [SecureDrop/SecureDrop Client](https://weblate.securedrop.org/projects/securedrop/securedrop-client/).

Current translation status:

Weblate translation status

### CODE REVIEW

File: securedrop_client/locale/bn/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=2; plural=n > 1;\n"
+"X-Generator: Weblate 4.13\n"
+"Generated-By: Babel 2.10.1\n"
+
+msgid "{application_name} is already running"
+msgstr "{application_name} চলমান"
+
+msgid "The SecureDrop server cannot be reached. Trying to reconnect..."
+msgstr "SecureDrop সার্ভারে পৌছানো যাচ্ছে না। পুনরায় চেষ্টা করা হচ্ছে..."
```
Comment (gonzalo-bulnes) [inline securedrop_client/locale/bn/LC_MESSAGES/messages.po:25]: Oh, we're gotta have to make sure we've got font support.

Incorrect rendering of Bengali text.

**Review by gonzalo-bulnes — COMMENTED** [review]
Fantastic! :tada: 

I suppose we'll let Weblate do its thing for a little bit while the translators add some strings. I'm glad to see a new language opening up. I'll keep the new requirements (see inline comment) in mind myself and put them on @tina-ux's radar.

File: securedrop_client/locale/bn/LC_MESSAGES/messages.po (hunk 1 above)
Comment (tina-ux) [reply securedrop_client/locale/bn/LC_MESSAGES/messages.po:25]: Thank you Gonzalo, you're right we should make sure we have font support! Adding this + localization guidelines in the UX wishlist

Comment (gonzalo-bulnes) [reply securedrop_client/locale/bn/LC_MESSAGES/messages.po:30]: Note for future reference: the [Noto](https://en.wikipedia.org/wiki/Noto_fonts) family looks very promising to me.

**Comment by gonzalo-bulnes** [comment]
After consulting with @cfm: we'll probably want to merge this PR before the next release, but there are no commitments to provide translations as part of the pilot program.

Unless there is something specifically wrong with this specific PR, I'd recommend we merge it. (I mean that if there are systemic issues, I'd still merge and flag them for further investigation.)

**Review by gonzalo-bulnes — CHANGES_REQUESTED** [review]
- [ ] `el` contains `fuzzy` translations that are not fit for purpose
- [ ]  `en-XA` has some translation without `!!!` exclamation marks
- [ ] `eo` machine file `.mo` has changed without changes in the corresponding `.po` (not sure if that's a problem, but it seems odd to me)
- [ ] There are changes in the `pt_BR` `.po` file without changes in the corresponding `.mo` file. (Same as above, not sure if that's wrong, but it seems odd to me.) Also happens in `ru`, `sk`, `tr`.
- [ ] `es`, `fi`, `hr`, `it`, `pt_BR`, `ru`, `sk`, `tr`, `zh_HANS` contain the same, or some inadequate `fuzzy`-matched strings
- [ ] `es` contains `Tile Capitalized Strings` that are not the way in Spanish, I've added context in the comment.

What do you think @cfm?

File: securedrop_client/locale/es/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -66,19 +70,34 @@ msgid "Failed to delete source at server"
...
+
 msgid "Download All Files"
-msgstr "Todos los archivos"
+msgstr "Descargar todos los archivos"
 
-#, fuzzy
-#| msgid "Deleting source account..."
 msgid "Delete Source Account"
-msgstr "Borrando cuenta de la fuente..."
+msgstr "Eliminar Cuenta de Fuente"
```
[the remark this answers is not in the document]
Comment (gonzalo-bulnes) [reply securedrop_client/locale/es/LC_MESSAGES/messages.po:94]: Note that `Borrando` is wrong, `Eliminar` is a good correction, but the title case is not a thing in Spanish. So a bit in wrong in both strings.

[the remark this answers is not in the document]
Comment (gonzalo-bulnes) [reply securedrop_client/locale/es/LC_MESSAGES/messages.po:94]: Note to self: I should sign up to translate into Spanish.

**Comment by cfm** [comment]
@gonzalo-bulnes, the problems you've listed in <https://github.com/freedomofpress/securedrop-client/pull/1548#pullrequestreview-1331808126> (thank you!) are indeed multiple, but I think most will turn out to have simple or harmless causes.  I agree with you, however, that one stands out as fundamental, blocking (of this pull request), and distractingly noisy: useless `fuzzy` translations per freedomofpress/securedrop#6483.  If we were looking for an overwhelming reason to disable fuzzy matching, this is it: a bad `fuzzy` translation prevents automatic fallback to the English source string, which is to say it's worse than having no translation at all.

I was only able to confirm this understanding with Localization Lab on Monday; I wasn't aware of this week's release goal for which we'd be considering this pull request.  Let's defer (omit) it from this release, and next week I'll revisit it fully after freedomofpress/securedrop#6483.

**Comment by gonzalo-bulnes** [comment]
Makes sense @cfm, let me know if you want a hand next week 👍

**Comment by gonzalo-bulnes** [comment]
**For the record**: @cfm and I took the decision to keep this PR out of the release scope for **0.9.0**, because there is currently no commitment to provide translations as part of the pilot program,and we think the issue described above should be addressed before we provide them. (It is preferable to provide incomplete, rather than potentially misleading translations.)

File: securedrop_client/locale/el/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+msgid "{application_name} is already running"
+msgstr "Το {application_name} εκτελείται ήδη"
+
+#, fuzzy
+#| msgid "Retrieving new messages"
+msgid "Retrieving 1 message"
+msgid_plural "Retrieving {message_count} messages"
+msgstr[0] "Λαμβάνονται τα νέα μηνύματα"
+msgstr[1] "Λαμβάνονται τα νέα μηνύματα"
```
[the remark this answers is not in the document]
Comment (cfm) [reply securedrop_client/locale/el/LC_MESSAGES/messages.po]: Fuzzy matching disabled per freedomofpress/securedrop#6483 and these entries manually cleared in Weblate.  On to the rest....

File: securedrop_client/locale/es/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -7,20 +7,27 @@ msgid ""
...
 msgid "{application_name} is already running"
 msgstr "{application_name} ya se está ejecutando"
 
+#, fuzzy
+#| msgid "Retrieving new messages"
+msgid "Retrieving 1 message"
+msgid_plural "Retrieving {message_count} messages"
+msgstr[0] "Descargando mensajes nuevos"
+msgstr[1] "Descargando mensajes nuevos"
+
```
[the remark this answers is not in the document]
Comment (cfm) [reply securedrop_client/locale/es/LC_MESSAGES/messages.po]: Fuzzy matching disabled per https://github.com/freedomofpress/securedrop/issues/6483 and these entries manually cleared in Weblate.

File: securedrop_client/locale/es/LC_MESSAGES/messages.po (hunk 2 above)
[the remark this answers is not in the document]
Comment (cfm) [reply securedrop_client/locale/es/LC_MESSAGES/messages.po]: Fuzzy matching disabled per https://github.com/freedomofpress/securedrop/issues/6483 and these entries manually cleared in Weblate.

File: securedrop_client/locale/eo/LC_MESSAGES/messages.mo
Comment (cfm) [inline securedrop_client/locale/eo/LC_MESSAGES/messages.mo:1]: This is deceptive: a by-product of #1598.  Filed upstream in WeblateOrg/Weblate#8936.

File: securedrop_client/locale/en_XA/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+msgstr "[!Failed to send!]"
+
+msgid " DOWNLOAD"
+msgstr " [ DOWNLOAD]"
+
+msgid "EXPORT"
+msgstr "[EXPORT]"
+
+msgid "PRINT"
+msgstr "[PRINT]"
```
[the remark this answers is not in the document]
Comment (cfm) [reply securedrop_client/locale/en_XA/LC_MESSAGES/messages.po:170]: This is the intended combination of fixed-length `[]` and variable-length `!` prefixes and suffixes:

> Using the variable parts makes it possible to look for strings which might not fit into the user interface after the localization - it extends the text based on the source string length. The variable parts are repeated by length of the text multiplied by the multiplier. For example `Hello world` with variable suffix `_` and variable multiplier of 1 becomes `Hello world___________` - the suffix is repeated once for each character in the source string.  (https://docs.weblate.org/en/weblate-4.14.1/admin/addons.html#pseudolocale-generation)

So `[]` flags the string as pseudolocalized, while the lack of `!` means the string is just short.  :-)

**Comment by cfm** [comment]
@gonzalo-bulnes, I've resolved or otherwise addressed your points in <https://github.com/freedomofpress/securedrop-client/pull/1548#pullrequestreview-1331808126>, including disabling fuzzy matching and manually removing fuzzy entries in existing translations.  The exception is <https://github.com/freedomofpress/securedrop-client/pull/1548#discussion_r1137885105>, which is awaiting an upstream fix or at least confirmation.

File: securedrop_client/locale/en_XA/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+msgstr "[!!!!!!!!!!!!!!!!!!!<h2>Some files will not be exported</h2>Some files from this source have not yet been downloaded, and will not be exported.<br /><br />To export the currently-downloaded files, click \"Continue.\"!!!!!!!!!!!!!!!!!!!]"
+
+msgid "all files and transcript"
+msgstr "[!!all files and transcript!!]"
+
+msgid "SecureDrop Client {}"
+msgstr "[!!SecureDrop Client {}!!]"
+
+msgid "Quit"
+msgstr "[Quit]"
```
Comment (gonzalo-bulnes) [inline securedrop_client/locale/en_XA/LC_MESSAGES/messages.po:113]: Not to be fixed here, rather in Webate. @cfm What's the workflow for this? Could I do it?

```suggestion
msgid "Quit"
msgstr "[!!Quit!!]"
```

File: securedrop_client/locale/fr/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+msgstr "Fichier : %(filename)s"
+
+msgid "Download All"
+msgstr ""
+
+msgid "Delete Source Account"
+msgstr "Supprimer le compte de la source"
+
+msgid "Delete All Files and Messages"
+msgstr "Supprimer tout les fichiers et messages"
```
Comment (gonzalo-bulnes) [inline securedrop_client/locale/fr/LC_MESSAGES/messages.po:92]:
```suggestion
msgstr "Supprimer tous les fichiers et messages"
```

:grimacing:

File: securedrop_client/locale/fr/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+msgstr ""
+
+msgid "Export All"
+msgstr ""
+
+msgid "<h2>Some files will not be exported</h2>Some files from this source have not yet been downloaded, and will not be exported.<br /><br />To export the currently-downloaded files, click \"Continue.\""
+msgstr ""
+
+msgid "all files and transcript"
+msgstr ""
```
Comment (gonzalo-bulnes) [inline securedrop_client/locale/fr/LC_MESSAGES/messages.po:107]: :thinking: I don't know where this string is coming from…

**Review by gonzalo-bulnes — DISMISSED** [review]
There are a few translations that I know for sure are wrong, but that should be fixed in Weblate, not here. In the spirit of continuous integration, I'd merge this PR.

File: securedrop_client/locale/en_XA/LC_MESSAGES/messages.po (hunk 6 above)
Comment (cfm) [reply securedrop_client/locale/en_XA/LC_MESSAGES/messages.po:113]: Per <https://github.com/freedomofpress/securedrop-client/pull/1548#discussion_r1137937192>, this is the pseduolocale generator working as designed: making both fixed- and variable-length (here, variable length = 0) additions.

File: securedrop_client/locale/fr/LC_MESSAGES/messages.po (hunk 8 above)
Comment (cfm) [reply securedrop_client/locale/fr/LC_MESSAGES/messages.po]: Looks like:

https://github.com/freedomofpress/securedrop-client/blob/aa6d07d5d2a06c27844095d635c1885e256604aa/securedrop_client/gui/actions.py#L340

File: securedrop_client/locale/fr/LC_MESSAGES/messages.po (hunk 7 above)
Comment (cfm) [reply securedrop_client/locale/fr/LC_MESSAGES/messages.po:92]: Yes: this is the kind of review Localization Lab's language teams will perform when we advise them of a release timeline.  The lesson we're learning here is that these automatic, cumulative "Translate update from Weblate" pull requests may at any time contain a mix of both reviewed and unreviewed translations, and the maintainer-reviewable `.po` diff does not indicate the Weblate review status.

File: securedrop_client/locale/en_XA/LC_MESSAGES/messages.po (hunk 5 above)
[the remark this answers is not in the document]
Comment (gonzalo-bulnes) [reply securedrop_client/locale/en_XA/LC_MESSAGES/messages.po:170]: Mmmm okay. Those modifications are for what I thought they were. :bulb: 

I thought they were to make non-internationalized strings obvious for a human reviewer, but they are instead to text variable word length scenarios. (That doesn't make it clear ot me how a 0-length addition is useful, but that's a detail.)

File: securedrop_client/locale/fr/LC_MESSAGES/messages.po (hunk 7 above)
Comment (gonzalo-bulnes) [reply securedrop_client/locale/fr/LC_MESSAGES/messages.po:92]: > The lesson we're learning here is that these automatic, cumulative "Translate update from Weblate" pull requests may at any time contain a mix of both reviewed and unreviewed translations, and the maintainer-reviewable `.po` diff does not indicate the Weblate review status.

Yes, exactly that!

**Review by gonzalo-bulnes — DISMISSED** [review]
With @cfm corrections and clarifications, I'd say :shipit: 

@cfm I let you judge on the timing for merging, since there are un-reviewed translations in there.

File: securedrop_client/locale/eo/LC_MESSAGES/messages.mo
Comment (cfm) [reply securedrop_client/locale/eo/LC_MESSAGES/messages.mo:1]: Fixed in 006a4a7ac8906910760ab13a1233231b9bd0483d.

**Pull request merged**